src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtFacilityService.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.Facility;
+import uk.gov.hmcts.dts.fact.repositories.CourtFacilityRepository;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+import uk.gov.hmcts.dts.fact.repositories.FacilityTypeRepository;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class AdminCourtFacilityService {
+    private final CourtRepository courtRepository;
+    private final CourtFacilityRepository courtFacilityRepository;
+    private final FacilityTypeRepository facilityTypeRepository;
+
+    @Autowired
+    public AdminCourtFacilityService(final CourtRepository courtRepository, final CourtFacilityRepository courtFacilityRepository, final FacilityTypeRepository facilityTypeRepository) {
+        this.courtRepository = courtRepository;
+        this.courtFacilityRepository = courtFacilityRepository;
+        this.facilityTypeRepository = facilityTypeRepository;
+    }
+
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    public List<Facility> getCourtFacilitiesBySlug(final String slug) {
+        return  courtRepository.findBySlug(slug)
+            .map(c -> c.getFacilities()
+                .stream()
+                .map(Facility::new)
+                .collect(toList()))
+            .orElseThrow(() -> new NotFoundException(slug));
+    }
+
+    @Transactional()
+    public List<Facility> updateCourtFacility(final String slug, final List<Facility> courtFacilities) {
+
+        final Court courtEntity = courtRepository.findBySlug(slug)
+            .orElseThrow(() -> new NotFoundException(slug));
+
+        return saveCourtFacilities(courtEntity,courtFacilities);
+    }
+
+    protected List<Facility> saveCourtFacilities(final Court courtEntity, final List<Facility> courtFacilities) {
+
+        final List<uk.gov.hmcts.dts.fact.entity.Facility> courtFacilitiesEntities = getNewCourtFacilityEntity(courtFacilities);
+
+        //remove existing court facilities and add updated facilities
+        courtFacilityRepository.deleteAll(courtEntity.getFacilities());
+        courtFacilityRepository.saveAll(courtFacilitiesEntities);
+
+        if (courtEntity.getFacilities().isEmpty()) {
+            courtEntity.setFacilities(courtFacilitiesEntities);
+        } else {
+            courtEntity.getFacilities().clear();
+            courtEntity.getFacilities().addAll(courtFacilitiesEntities);
+        }
+
+        courtRepository.save(courtEntity);
+
+        return courtEntity.getFacilities().stream()
```

Comment:
I am a bit confused as why is there a need to have 2 separate call to update the database. I thought the Court entity already linked to the search_facility table through the search_courtfacility table. So if you save to the CourtRepository, shouldn't it automatically update the underlying joint tables?

Reply:
yeah originally that's what I thought but its not updating the search_facility table so adding in the extra call

Reply:
Adam's areas of law stuff is very similar to your facility stuff. If you look at his PR https://github.com/hmcts/fact-api/pull/420/files, you can see that he only calls the repository once to update the values. The difference is his CourtAreaOfLawRepository is operating on the joining entity CourtAreaOfLaw, while your CourtFacilityRepository is not operating on the joining entity CourtFacility but on Facility entity directly.

Reply:
I have updated the service using only the CourtFacilityRepository to update court facilities which is now operating on CourtFacility entity